src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -213,7 +228,16 @@ public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, in
 
             using (OpenNewDocumentStateScope(options ?? workspace.Options))
             {
-                var document = OpenDocument(workspace, documentId);
+                // Before attempting to open the document, check if the location maps to a different file that should be opened instead.
+                var document = workspace.CurrentSolution.GetDocument(documentId);
+                var spanMappingService = document?.Services.GetService<ISpanMappingService>();
+                if (spanMappingService != null)
+                {
+                    var textSpan = getTextSpanForMapping(document);
+                    return TryNavigateToMappedFile(workspace, spanMappingService, document, textSpan);
```

Comment:
I'm guessing `document.Services.GetService` will only return a service if the file in question is mapped? Because otherwise, it seems odd that when `TryNavigateToMappedFile` returns false, we wouldn't try to open the file as a normal file.

Reply:
yup, from my testing the document mapping service is only returned for documents that are mapped.  Since documents that have mapped files can't be opened via the normal workspace open methods, it doesn't make sense to try.